src/devices/acurite.c

```diff
@@ -662,6 +731,106 @@ static int acurite_txr_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+            data = data_make(
+                    "model",            "",         DATA_STRING, "Acurite-Atlas",
+                    "id",               NULL,       DATA_INT,    sensor_id,
+                    "channel",          NULL,       DATA_STRING, &channel_str,
+                    "sequence_num",     NULL,       DATA_INT,    sequence_num,
+                    "battery_ok",       NULL,       DATA_INT,    !battery_low,
+                    "message_type",     NULL,       DATA_INT,    message_type,
+                    "wind_avg_mi_h",    "Wind Speed", DATA_FORMAT, "%.1f mi/h", DATA_DOUBLE, wind_speed_mph,
+                    NULL);
+            /* clang-format on */
+
+            if (message_type == ACURITE_MSGTYPE_ATLAS_WINDSPEED_TEMP_HUMIDITY) {
+                // Wind speed, temperature and humidity
+
+                // range -40 to 160 F
+                // FIXME: are there really 13 bits? use 11 for now.
+                int temp_raw = (bb[4] & 0x0F) << 7 | (bb[5] & 0x7F);
+                tempf = (temp_raw - 400) * 0.1;
+
+                humidity = (bb[6] & 0x7f); // 1-99 %rH
+
+                /* clang-format off */
+                data = data_append(data,
+                        "temperature_F",    "temperature",  DATA_FORMAT,    "%.1f F",       DATA_DOUBLE, tempf,
+                        "humidity",         NULL,           DATA_FORMAT,    "%u %%",        DATA_INT,    humidity,
+                        NULL);
+                /* clang-format on */
+            }
+
+            if (message_type == ACURITE_MSGTYPE_ATLAS_WINDSPEED_RAINFALL) {
+                // Wind speed, wind direction, and rain fall
+                wind_dir = ((bb[4] & 0x1f) << 5) | ((bb[5] & 0x7c) >> 2);
+
+                // range: 0 to 99.99 in, 0.01 inch increments, accumulated
+                // FIXME: are there really only 7 bits? use 9 for now.
+                raincounter = ((bb[5] & 0x03) << 7) | (bb[6] & 0x7F);
+
+                /* clang-format off */
+                data = data_append(data,
+                        "wind_dir_deg",     NULL,           DATA_FORMAT,    "%.1f",         DATA_DOUBLE, wind_dir,
+                        "rain_in",          "Rainfall Accumulation", DATA_FORMAT, "%.2f in", DATA_DOUBLE, raincounter * 0.01f,
+                        NULL);
+                /* clang-format on */
+            }
+
+            if (message_type == ACURITE_MSGTYPE_ATLAS_WINDSPEED_UV_LUX) {
+                // Wind speed, UV Index, Light Intensity, Lightning?
+                int uv = (bb[4] & 0x3f);
+                int lux = ((bb[5] & 0x7f) << 7) | (bb[6] & 0x7F);
+
+                /* clang-format off */
+                data = data_append(data,
+                        "uv",               NULL,           DATA_INT, uv,
+                        "lux",              NULL,           DATA_INT, lux * 10,
+                        NULL);
+                /* clang-format on */
+            }
+
+            int byte8  = (bb[7] & 0x7f);
```

Comment:
If handling the shorter messages without the lightning detector, need to make sure this is safe to do first.

Reply:
This returns useless data when the message length is too short (when no LD is installed). This should have some message type logic applied to it.